s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/AWSV2SourceClient.java

```diff
@@ -79,46 +83,52 @@ public AWSV2SourceClient(final S3SourceConfig s3SourceConfig, final Set<String>
     }
 
     public Iterator<String> getListOfObjectKeys(final String startToken) {
-        final ListObjectsV2Request request = new ListObjectsV2Request().withBucketName(bucketName)
-                .withMaxKeys(s3SourceConfig.getS3ConfigFragment().getFetchPageSize() * PAGE_SIZE_FACTOR);
-
-        if (StringUtils.isNotBlank(startToken)) {
-            request.withStartAfter(startToken);
-        }
-        // Prefix is optional so only use if supplied
-        if (StringUtils.isNotBlank(s3SourceConfig.getAwsS3Prefix())) {
-            request.withPrefix(s3SourceConfig.getAwsS3Prefix());
-        }
+        final ListObjectsV2Request request = ListObjectsV2Request.builder()
+                .bucket(bucketName)
+                .maxKeys(s3SourceConfig.getS3ConfigFragment().getFetchPageSize() * PAGE_SIZE_FACTOR)
+                .prefix(optionalKey(s3SourceConfig.getAwsS3Prefix()))
+                .startAfter(optionalKey(startToken))
+                .build();
 
         final Stream<String> s3ObjectKeyStream = Stream
                 .iterate(s3Client.listObjectsV2(request), Objects::nonNull, response -> {
                     // This is called every time next() is called on the iterator.
                     if (response.isTruncated()) {
-                        return s3Client.listObjectsV2(
-                                new ListObjectsV2Request().withContinuationToken(response.getNextContinuationToken()));
+                        return s3Client.listObjectsV2(ListObjectsV2Request.builder()
+                                .maxKeys(s3SourceConfig.getS3ConfigFragment().getFetchPageSize() * PAGE_SIZE_FACTOR)
```

Comment:
Do we need to have maxKeys here, just for my understanding, as we don't have prefix check.?

Reply:
I think the maxkeys is still required here but I would probably need to test it to see if it remembers the original max keys setting as I could not find that info in the documentation.